Switched to dune

ttamttam opened the pull request.

ttamttam committed:
Switched to dune

olafhering commented:
I approve the concept. But this single change seems to mix fixes with the switch to dune? Perhaps it is worth splitting code changes from the buildsystem change.

ttamttam commented:
I'll have a look at it…

ttamttam committed:
Synchronized "Makefile" and "facile.opam" commands

ttamttam committed:
Obsolete "Pervasives" calls replaced

"stdlib-shims" dependency is added to assure backward compatibility for older
compilers.

ttamttam committed:
Switch to dune

The examples compile with: `dune build profile=release`

ttamttam committed:
Examples fix: obsolete Array.create becomes Array.make

ttamttam committed:
Examples: renamed some unused variables to avoid warnings

ttamttam committed:
Examples: adapted 'scheduling.mli' formatting to avoid warnings

ttamttam committed:
Revert "Switched to dune"

This reverts commit 0f802ce9cc356b511802ab6bcfbec9be1cbf3d62.

UnixJunkie commented:
This PR should be merged.

UnixJunkie commented:
@olafhering what is the lower bound version of ocaml that you want to stay compatible with?

olafhering commented:
4.14 is fine with me, this is what will be included in `Leap 15.6`.

UnixJunkie commented:
Superseded by https://github.com/Emmanuel-PLF/facile/pull/4
But honestly, all the changes were already proposed in this PR.
